### ISSUE TITLE
Changing GIT repo results in error

### ISSUE THREAD

**Issue by markdjones82** [aws-ia/terraform-aws-control_tower_account_factory#184]
**Terraform Version & Prov:**
TFE, CLI 1.1.0

**AFT Version:**
1.3.5

**Terraform Version & Provider Versions**
Please provide the outputs of `terraform version` and `terraform providers` from within your AFT environment

`terraform version`
```
1.1.7
```

`terraform providers`
```
AFT
```

**Bug Description**
Changing the Git enterprise host address errors out with a regex issue.

**To Reproduce**
1. Update AFT module call variable: github_enterprise_url 
2. Run Plan and Apply.  Recieve error below.
3. 
Error: error updating CodeStar Connections Host (arn:aws:codestar-connections:us-west-2:<accountID:host/github-ent-8c30a7d3): ValidationException: 1 validation error detected: Value 'github-enterprise-host' at 'hostArn' failed to satisfy constraint: Member must satisfy regular expression pattern: arn:aws(-[\w]+)*:codestar-connections:.+:[0-9]{12}:host\/.+ status code: 400, request id: 8be33d55-69e4-4dff-b99b-8be14697e977
with module.aft.module.aft_code_repositories.aws_codestarconnections_host.githubenterprise[0]
on .terraform/modules/aft/modules/aft-code-repositories/codestar.tf line 22, in resource "aws_codestarconnections_host" "githubenterprise":
resource "aws_codestarconnections_host" "githubenterprise" {

**Expected behavior**
Changing hit host should update.

**Related Logs**
Provide any related logs or error messages to help explain your problem.

**Additional context**

**Comment by markdjones82** [comment]
In addition to this, the CodePipelines do not use the new codestar connection.  The only fix I found was to run a manual code build on aft-create-pipeline and pass the VENDED_ACCOUNT_ID to the environment variables.  This updated the codepipeline.  Is there a better way to do this?

**Comment by markdjones82** [comment]
> As you noted you will need to force an update to the Pipeline creation. A more simple way is to resubmit the account request with a change to something like `change_management_parameters` which have no impact to downstream events but should enforce an update to the DDB and invoke the trigger. The standard process will kick off and in my case I deleted the pipeline so it was recognized as missing and recreated. It might also work without deleting the pipeline but I haven't looked to see if the code has a test to skip pipeline creation if it finds one.

Yeah, I actually tried that as well, but it doesn't update existing pipeline.  I hadn't thought about deleting the pipeline.

**Comment by balltrev** [comment]
@markdjones82 I have gone ahead and opened a backlog item with the team to better handle this use case. Thanks for the details here!

**Comment by markdjones82** [comment]
> @markdjones82 I have gone ahead and opened a backlog item with the team to better handle this use case. Thanks for the details here!

Cool thanks. Overall like the product! Have a few other suggestions if you all ever want to chat about it :)
